src/mon/MonClient.cc

```diff
@@ -1368,3 +1271,93 @@ void MonClient::register_config_callback(md_config_t::config_callback fn) {
 md_config_t::config_callback MonClient::get_config_callback() {
   return config_cb;
 }
+
+class monc_error_category : public ceph::converting_category {
+public:
+  monc_error_category(){}
+  const char* name() const noexcept override;
+  std::string message(int ev) const noexcept override;
+  boost::system::error_condition default_error_condition(int ev) const noexcept
+    override;
+  bool equivalent(int ev, const boost::system::error_condition& c) const
+    noexcept override;
+  using ceph::converting_category::equivalent;
+  int from_code(int ev) const noexcept override;
+};
+
+const char* monc_error_category::name() const noexcept {
+  return "monc";
+}
+
+std::string monc_error_category::message(int ev) const noexcept {
+  using namespace ::monc_errc;
+
+
+  switch (ev) {
+  case 0:
+    return "No error";
+
+  case shutting_down: // Command failed due to MonClient shutting down
+    return "Command failed due to MonClient shutting down";
+  case session_reset:
+    return "Monitor session was reset";
+  case rank_dne:
+    return "Requested monitor rank does not exist";
+  case mon_dne:
+    return "Requested monitor does not exist";
+  case timed_out:
+    return "Monitor operation timed out";
+  }
+
+  return "Unknown error";
+}
+
+boost::system::error_condition monc_error_category::default_error_condition(int ev) const noexcept {
+  using namespace ::monc_errc;
+  switch (ev) {
+  case shutting_down:
+    return boost::system::errc::operation_canceled;
+  case session_reset:
+    return boost::system::errc::resource_unavailable_try_again;
+  case rank_dne:
```

Comment:
Please consider using attribute [[fallthrough]].

Reply:
I would, but it doesn't seem to be allowed in this case. When I try:

```
case rank_dne:
  [[fallthrough]]
case monc_errc::mon_dne:
  return ceph::errc::not_in_map;
```

It flags `[[fallthrough]]` with ` "fallthrough attribute is only allowed on empty statements" error.

(But if you know what I'm doing wrong, I'll change it.)

Reply:
Hm! I haven't yet been able to reproduce this with either clang or gcc! It looks to me like what you're doing should be ok. :-/

Reply:
Done!

Reply:
it looks like [[fallthrough]] should have a trailing semicolon. see https://godbolt.org/z/5Izb60 for an example. with clang (trunk):
```
<source>:10:11: error: fallthrough attribute is only allowed on empty statements
        [[fallthrough]]
          ^
<source>:10:24: note: did you forget ';'?
        [[fallthrough]]
                       ^
                       ;
```
with gcc (trunk):
```
<source>: In function 'int main(int, char**)':
<source>:11:22: warning: 'fallthrough' attribute ignored [-Wattributes]
   11 |         case foo::baz:
      |                      ^
```

Reply:
@cbodley Ah! That's it!